### PR TITLE
stdlib: Add Intel HEX support to InstallRawStep

### PR DESCRIPTION
**Pull request by ehaas**
This allows writing HEX files with `exe.installRaw`, where `exe` is a `LibExeObjStep`. A HEX file will be written if the file extension is `.hex` or `.ihex`, otherwise a binfile will be written. The output format can be explicitly chosen with `exe.installRawWithFormat("filename", .hex);` (or `.bin`)

Part of #2826

### CODE REVIEW

**Comment by ehaas** [comment]
Drone failed on `Test [193/1032] behavior.atomics.test "behavior-native-ReleaseFast-bare-single atomicrmw with floats"... FAIL (TestUnexpectedResult)` which seems to be unrelated.

File: lib/std/build/InstallRawStep.zig [hunk 1]
```diff
@@ -13,6 +13,7 @@ const fs = std.fs;
 const io = std.io;
 const sort = std.sort;
 const warn = std.debug.warn;
+const nl = std.cstr.line_sep;
```
Comment (andrewrk) [inline lib/std/build/InstallRawStep.zig:16]: shouldn't the newline depend on the target, not the host system running the compiler?

**Review by andrewrk — COMMENTED** [review]
Thanks for this! Looks merge-ready to me except for 1 question that I had-

File: lib/std/build/InstallRawStep.zig (hunk 1 above)
Comment (ehaas) [reply lib/std/build/InstallRawStep.zig:16]: Upon further investigation it appears to always end in CRLF. Source: https://developer.arm.com/documentation/ka003292/latest and also running `objcopy -O ihex test.o test.hex` on a native .o file for a x86_64 linux system. So I'll update it to always to `\r\n`

**Pull request merged**